zenodo/modules/github/cli.py

```diff
@@ -0,0 +1,348 @@
+from invenio_accounts.models import User
+from invenio_db import db
+from invenio_github.api import GitHubAPI
+from invenio_github.models import Repository
+from sqlalchemy.orm.exc import NoResultFound
+
+
+@click.group()
+def github():
+    """Management commands for GitHub."""
+
+
+def user_param(ctx, param, value):
+    """Resolve user from CLI input parameter (email or User ID)."""
+    if value is None:
+        return
+    if '@' in value:  # Try to resolve as email
+        try:
+            user = User.query.filter_by(email=value.strip()).one()
+            # gha = GitHubAPI(user_id=user.id)
+            # if gha.api.me().email != user.email:
+
+        except NoResultFound:
+            raise Exception("User {0} not found.".format(value))
+    else:
+        try:
+            user_id = int(value)
+            user = User.query.get(user_id)
+        except ValueError:
+            raise Exception("User parameter must be either an email or ID")
+        if user is None:
+            raise Exception("User {0} not found.".format(value))
+    return user
+
+
+def repo_param(ctx, param, value):
+    """Resolve GitGub repository CLI input parameter (name or GitHub ID)."""
+    if value is None:
+        return
+    try:
+        if '/' in value:
+            repo = Repository.query.filter_by(name=value).one()
+        else:
+            try:
+                github_id = int(value)
+            except ValueError:
+                raise Exception('GitHub repository parameter must be either'
+                                ' a name or GitHub ID')
+            repo = Repository.query.filter_by(github_id=github_id).one()
+    except NoResultFound:
+        raise Exception("Repository {0} not found.".format(value))
+    return repo
+
+
+def repo_param_multiple(ctx, param, value):
+    """Resolve GitHub repository param with 'multiple' flag."""
+    return tuple(repo_param(ctx, param, v) for v in value)
+
+
+def verify_email(user):
+    """Check if user's GitHub and account emails match."""
+    gha = GitHubAPI(user_id=user.id)
+    gh_email = gha.api.me().email
+    if gh_email == user.email:
+        click.confirm("Warning: User's GitHub email ({0}) does not match"
+                      " the account's ({1}). Continue?".format(gh_email,
+                                                               user.email),
+                      abort=True)
+
+
+@github.group()
+def repo():
+    """Repository management commands."""
+
+
+@repo.command()
+@click.argument('user', callback=user_param)
+@click.option('--sync', '-s', default=False, is_flag=True,
+              help='Sync the repository prior to listing.')
+@click.option('with_all', '--all', '-a', default=False, is_flag=True,
+              help='List all repositories available.')
+@click.option('skip_email', '--skip-email-verification', '-E', is_flag=True,
+              help="Skip GitHub email verification.")
+@with_appcontext
+def list(user, sync, with_all, skip_email):
```

Comment:
Please rename to avoid overload builtin function (command can still be named list).